### PR TITLE
[multistage] clean up transform function type

### PR DESCRIPTION
**Pull request by walterddr**
registration of the transform function is not complete with `_` separate and camelCase naming. 

- [x] making the list of transform function complete
- [x] making sure there's a scalar function equivalent (or explicitly document when there's not)
- [x] add function registration test

TODO: add tests to pinot-query-runtime (separate b/c there are bug fixes required)

CC @xiangfu0 @Jackie-Jiang

### CODE REVIEW

File: pinot-common/src/test/java/org/apache/pinot/common/function/FunctionDefinitionRegistryTest.java [hunk 1]
```diff
@@ -47,10 +47,8 @@ public class FunctionDefinitionRegistryTest {
       // Scalar function
       "scalar",
       // Functions without scalar function counterpart as of now
-      // TODO: Revisit this list when we add scalar function for these functions
-      "arraylength", "arrayaverage", "arraymin", "arraymax", "arraysum",
-      "valuein", "mapvalue", "inidset", "lookup", "groovy",
-      "timeconvert", "datetimeconvert", "jsonextractscalar", "jsonextractkey", "clpdecode",
+      "arraylength", "arrayaverage", "arraymin", "arraymax", "arraysum", "clpdecode", "groovy",
```
Comment (walterddr) [inline pinot-common/src/test/java/org/apache/pinot/common/function/FunctionDefinitionRegistryTest.java:50]: other than the ignored list here, all other functions (including their alternative naming are registered in FunctionRegistry (e.g. has correspondent in ScalarFunction impl)

File: pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java [hunk 2]
```diff
@@ -70,25 +70,27 @@ public enum TransformFunctionType {
 
   // predicate functions
   EQUALS("equals"),
-  NOT_EQUALS("not_equals"),
-  GREATER_THAN("greater_than"),
-  GREATER_THAN_OR_EQUAL("greater_than_or_equal"),
+  NOT_EQUALS("not_equals", "notEquals"),
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java:73]: Do we need these aliases for auto-generated functions? If so, why skipping `LESS_THAN` and `NOT_IN`?

Comment (walterddr) [reply pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java:73]: lol good catch. :-)

Comment (walterddr) [reply pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java:73]: technically we dont need any of these. but we already have ComparisonFunctions in scalar with these names so added `lessThan` but not `notIn` b/c we dont have that

Comment (Jackie-Jiang) [reply pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java:73]: This is not the correct way to decide whether to add them though. Eventually we need scalar function for all of them, but that is orthogonal to this problem.
Does postgreSQL support these functions? If so, we need to add them, if not, we should not add them

Comment (walterddr) [reply pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java:73]: So 2 options. Either we allow underscore on both scalar and transform. Or we don't allow them in neither.

Problem is that transform function in v1 allows underscore so for backward compatibility we need to register them. It is not about syntax. 

For syntax sake we should not allow underscore format all together.

Comment (walterddr) [reply pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java]: discussed offline. will remove these in separate PR as we should not allow these function explicitly being called. They are not consider SqlFunction but SqlOperators to begin with

due to the fact that we do allow these already, we will use a separate PR to deprecate them

**Review by Jackie-Jiang — APPROVED** [review]

File: pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java (hunk 2 above)
Comment (walterddr) [reply pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java]: see #11734

**Pull request merged**